### PR TITLE
Added caching decorator

### PR DESCRIPTION
**Pull request by albert-nowak**
**JIRA ticket:** [WDP-137](https://kindergarten.stxnext.pl/browse/WDP-137)

### CODE REVIEW

File: src/presence_analyzer/utils.py [hunk 1]
```diff
@@ -34,6 +39,36 @@ def inner(*args, **kwargs):
...
+            global CACHE
+            now = time()
+            try:
+                if (now - CACHE[function]['last_caching']) > period:
+                    CACHE[function]['data'] = function(*args, **kwargs)
+                    CACHE[function]['last_caching'] = time()
+            except KeyError:
+                with lock:
+                    CACHE.setdefault(function, {})['data'] = function(*args,
+                                                                      **kwargs)
```
Comment (albert-nowak) [inline src/presence_analyzer/utils.py]: Not sure if this is correct split of this line (one line is too long)

File: src/presence_analyzer/tests.py [hunk 2]
```diff
@@ -254,6 +254,16 @@ def test_work_hours(self):
...
 
+    def test_cache(self):
+        """
+        Test caching data in memory for given time.
+        """
+        data = utils.get_data_from_csv()
+        self.assertDictEqual(
+            data,
+            utils.CACHE['get_data_from_csv']['data']
+        )
```
Comment (albert-nowak) [inline src/presence_analyzer/tests.py]: I feel like i should test more cases but don't know which

File: src/presence_analyzer/utils.py [hunk 3]
```diff
@@ -34,6 +39,38 @@ def inner(*args, **kwargs):
...
+        },
+    }
+    """
+    def decorating_wrapper(function):
+        def caching_wrapper(*args, **kwargs):
+            global CACHE
+            func_name = function.__name__
+            try:
+                if (time() - CACHE[func_name]['last_caching']) > period:
+                    CACHE[func_name]['data'] = function(*args, **kwargs)
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]:
```
CACHE[func_name] = {
    'data': funtion(*args, **kwargs),
    'last_caching': time()
}
```

File: src/presence_analyzer/utils.py [hunk 4]
```diff
@@ -8,6 +8,8 @@
 from datetime import datetime
 from functools import wraps
 from json import dumps
+from time import time
+from threading import Lock
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: sort imports

**Pull request merged**